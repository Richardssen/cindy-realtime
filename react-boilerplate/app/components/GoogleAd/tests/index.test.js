import React from 'react';
import { shallow } from 'enzyme';

import { RegisterGoogleAd, GoogleAd } from '../index';

const FakeGoogleAd = () => <div id="fakeGoogleAd" />;
const fakeToken = 'fakeToken';
const fakeSlot = '000000';
const fakeFormat = 'auto';

describe('<GoogleAd />', () => {
  describe('RegisterGoogleAd', () => {
    it('renders null if clientToken is null', () => {
      const Element = RegisterGoogleAd({ clientToken: null })(FakeGoogleAd);
      const rendered = shallow(<Element />);
      expect(rendered.contains(<FakeGoogleAd />)).toEqual(false);
    });

    it('renders null if clientToken is undefined', () => {
      const Element = RegisterGoogleAd({})(FakeGoogleAd);
      const rendered = shallow(<Element />);
      expect(rendered.contains(<FakeGoogleAd />)).toEqual(false);
    });

    it('renders GoogleAd if clientToken is defined', () => {
      const Element = RegisterGoogleAd({ clientToken: fakeToken })(
        FakeGoogleAd
      );
      const rendered = shallow(<Element />);
      expect(rendered.contains(<FakeGoogleAd client={fakeToken} />)).toEqual(
        true
      );
    });
  });
  describe('GoogleAd', () => {
    it('Expect to accept required props', () => {
      window.adsbygoogle = [];
      expect(window.adsbygoogle.length).toBe(0);
      const rendered = shallow(
        <GoogleAd client={fakeToken} slot={fakeSlot} format={fakeFormat} />
      );
      expect(window.adsbygoogle.length).toBe(1);
      expect(rendered).toMatchSnapshot();
    });
    it('Expect to accept all props', () => {
      const rendered = shallow(
        <GoogleAd
          client={fakeToken}
          slot={fakeSlot}
          format={fakeFormat}
          wrapperDivStyle={{ width: '100%' }}
          layoutKey="-gw-3+1f-3d+2z"
        />
      );
      expect(rendered).toMatchSnapshot();
    });
  });
});
