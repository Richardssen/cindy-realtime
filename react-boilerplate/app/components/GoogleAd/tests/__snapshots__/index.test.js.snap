// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GoogleAd /> GoogleAd Expect to accept all props 1`] = `
<div
  style={
    Object {
      "width": "100%",
    }
  }
>
  <ins
    className="adsbygoogle"
    data-ad-client="fakeToken"
    data-ad-format="auto"
    data-ad-layout-key="-gw-3+1f-3d+2z"
    data-ad-slot="000000"
    style={
      Object {
        "display": "block",
      }
    }
  />
</div>
`;

exports[`<GoogleAd /> GoogleAd Expect to accept required props 1`] = `
<div
  style={
    Object {
      "overflow": "hidden",
    }
  }
>
  <ins
    className="adsbygoogle"
    data-ad-client="fakeToken"
    data-ad-format="auto"
    data-ad-slot="000000"
    style={
      Object {
        "display": "block",
      }
    }
  />
</div>
`;
